src/util/locking/VoidLocker.ts

```diff
@@ -0,0 +1,33 @@
+import type { ResourceIdentifier } from '../../http/representation/ResourceIdentifier';
+import { getLoggerFor } from '../../logging/LogUtil';
+import type { ExpiringReadWriteLocker } from './ExpiringReadWriteLocker';
+/**
+ * This locker will execute the whileLocked function without any locking mechanism
+ *
+ * Do not use this locker in combination with storages that doesn't handle concurrent read/writes gracefully
+ */
+export class VoidLocker implements ExpiringReadWriteLocker {
+  protected readonly logger = getLoggerFor(this);
+
+  public constructor() {
+    this.logger.warn('Warning: this locker will execute the whileLocked function without any locking mechanism');
+  }
+
+  public async withReadLock<T>(
+    identifier: ResourceIdentifier,
+    whileLocked: (maintainLock: () => void) => T | Promise<T>,
+  ): Promise<T> {
+    return whileLocked((): void => {
+      // eslint:disable-next-line:no-empty
+    });
```

Comment:
```suggestion
    return whileLocked(noop);
```

with `noop` a constant defined on top.

Reply:
Thanks for your suggestions.